snuba/utils/streams/backends/kafka.py

```diff
@@ -615,26 +617,100 @@ def closed(self) -> bool:
 
 DEFAULT_QUEUED_MAX_MESSAGE_KBYTES = 50000
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
+DEFAULT_PARTITIONER = "consistent"
+DEFAULT_MAX_MESSAGE_BYTES = 50000000  # 50MB, default is 1MB
+SUPPORTED_KAFKA_CONFIGURATION = (
+    # Check https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    # for the full list of available options
+    "bootstrap.servers",
+    "sasl.mechanism",
+    "sasl.username",
+    "sasl.password",
+    "security.protocol",
+)
+DEFAULT_STORAGE_NAME = ""
+
+
+def get_default_kafka_configuration(
+    storage_name: Optional[str] = None,
+    bootstrap_servers: Optional[Sequence[str]] = None,
+    override_params: Optional[Mapping[str, Any]] = None,
+) -> KafkaBrokerConfig:
+    if storage_name in settings.DEFAULT_STORAGE_BROKERS:
+        # this is now deprecated
+        logger.warning(
+            "DEPRECATED: DEFAULT_STORAGE_BROKERS is defined. Please use STORAGE_BROKER_CONFIG instead"
+        )
+        default_config = {}
+        bootstrap_servers = ",".join(settings.DEFAULT_STORAGE_BROKERS[storage_name])
+    else:
+        default_config = settings.STORAGE_BROKER_CONFIG.get(
```

Comment:
I think this is missing the case where the user leaves `DEFAULT_STORAGE_BROKERS` empty and overrides `DEFAULT_BROKERS`. If the user does that we would end up using `BROKER_CONFIG` instead of `DEFAULT_BROKERS` which is wrong. 

I would advise you wrote a unit test for this logic. Overriding settings for unit tests is trivial and it would be very easy to ensure all cases are addressed in the right order:
- the user overrides one of the legacy variable
- the user overrides both legacy variables
- the user does not override legacy variables and relies on default
- the user does not override legacy variables but overrides the new ones

Reply:
good call, yes let me add the unit test and fix this function

Reply:
updated and unittest added